drivers/platform/mbed/mbed_irq.cpp

```diff
@@ -0,0 +1,569 @@
+
+	if (!desc || !desc->extra || !callback_desc)
+		return -EINVAL;
+
+	switch (irq_id) {
+	case EXTERNAL_INT_ID1:
+	case EXTERNAL_INT_ID2:
+	case EXTERNAL_INT_ID3:
+	case EXTERNAL_INT_ID4:
+	case EXTERNAL_INT_ID5:
+		switch (irq_id) {
+		case EXTERNAL_INT_ID1:
+			mbed_irq_callbacks.callback_ext_int_id1.callback = callback_desc->callback;
+			mbed_irq_callbacks.callback_ext_int_id1.ctx = callback_desc->ctx;
+			mbed_callback = mbed_ext_int_id1_callback;
+			break;
+
+		case EXTERNAL_INT_ID2:
+			mbed_irq_callbacks.callback_ext_int_id2.callback = callback_desc->callback;
+			mbed_irq_callbacks.callback_ext_int_id2.ctx = callback_desc->ctx;
+			mbed_callback = mbed_ext_int_id2_callback;
+			break;
+
+		case EXTERNAL_INT_ID3:
+			mbed_irq_callbacks.callback_ext_int_id3.callback = callback_desc->callback;
+			mbed_irq_callbacks.callback_ext_int_id3.ctx = callback_desc->ctx;
+			mbed_callback = mbed_ext_int_id3_callback;
+			break;
+
+		case EXTERNAL_INT_ID4:
+			mbed_irq_callbacks.callback_ext_int_id4.callback = callback_desc->callback;
+			mbed_irq_callbacks.callback_ext_int_id4.ctx = callback_desc->ctx;
+			mbed_callback = mbed_ext_int_id4_callback;
+			break;
+
+		case EXTERNAL_INT_ID5:
+			mbed_irq_callbacks.callback_ext_int_id5.callback = callback_desc->callback;
+			mbed_irq_callbacks.callback_ext_int_id5.ctx = callback_desc->ctx;
+			mbed_callback = mbed_ext_int_id5_callback;
+			break;
+
+		default:
+			return -EINVAL;
+		}
+
+		ext_interrupt = (InterruptIn *)(((struct mbed_irq_desc *)(
+				desc->extra))->int_obj);
+
+		/* Select interrupt mode */
+		if (((struct mbed_irq_desc *)(desc->extra))->int_mode == EXT_IRQ_FALL)
+			ext_interrupt->fall(mbed_callback);
+		else if (((struct mbed_irq_desc *)(desc->extra))->int_mode == EXT_IRQ_RISE)
+			ext_interrupt->rise(mbed_callback);
+		else
+			return -EINVAL;
+
+		break;
+
+	case TICKER_INT_ID:
+		/* Callback is attached from 'mbed_irq_enable' as ticker attach immediately
+		 * start invoking the interrupts */
+		mbed_irq_callbacks.callback_ticker_id.callback = callback_desc->callback;
+		mbed_irq_callbacks.callback_ticker_id.ctx = callback_desc->ctx;
+		break;
+
+	case UART_RX_INT_ID1:
+		/* Callback is attached from 'mbed_irq_enable' as uart attach immediately
+		 * start invoking the interrupts */
+		mbed_irq_callbacks.callback_uart_rx_id1.callback = callback_desc->callback;
+		mbed_irq_callbacks.callback_uart_rx_id1.ctx = callback_desc->ctx;
+		break;
+
+	default:
+		return -EINVAL;
+	}
+
+	return SUCCESS;
+}
+
+/**
+ * @brief	Unregister a IRQ callback function.
+ * @param	desc[in] - The IRQ controller descriptor.
+ * @param	irq_id[in] - Interrupt identifier.
+ * @return	SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t mbed_irq_unregister(struct irq_ctrl_desc *desc, uint32_t irq_id)
+{
+	if (!desc || !desc->extra)
+		return -EINVAL;
+
+	switch (irq_id) {
+	case  EXTERNAL_INT_ID1:
+		mbed_irq_callbacks.callback_ext_int_id1.callback = NULL;
```

Comment:
We should call `ext_interrupt->rise/fall(NULL);` here to unregister in the DFP as well.